File: 26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
    public int removeDuplicates(int[] nums) {

        int n = nums.length;

        int read = 0;
        int write = 1;

        while (read < n-1) {
            if (nums[read] != nums[read+1]) nums[write++] = nums[read+1];

            read++;
        }

        return write;
        
    }
}
// @lc code=end

File: 26.remove-duplicates-from-sorted-array.c
/*
 * @lc app=leetcode id=26 lang=c
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start


int removeDuplicates(int* nums, int numsSize){

    if (numsSize < 1) return 0;

    int read = 0, write = 0;

    while (read < numsSize) {
        if (nums[read] != nums[write]) {
            nums[++write] = nums[read];
        }

        read++;
    }

    return write + 1;



}
// @lc code=end

